--- a/whisper/utils.py
+++ b/whisper/utils.py
@@ -45,4 +45,4 @@
         for segment in result["segments"]:
             print(round(1000 * segment["start"]), file=file, end="\t")
             print(round(1000 * segment["end"]), file=file, end="\t")
-            print(segment["text"].strip().replace("\t", " "), file=file, flush=True)
+            print(segment["text"].strip().replace("\t", " ").replace("\r\n", " ").replace("\r", " ").replace("\n", " "), file=file, flush=True)
